File: calipso-hub-webapp/src/main/webapp/js/model/text.js
/*
 * Copyright (c) 2007 - 2013 www.Abiss.gr
 *
 * This file is part of Calipso, a software platform by www.Abiss.gr.
 *
 * Calipso is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Calipso is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with Calipso. If not, see http://www.gnu.org/licenses/agpl.html
 */
define([ 'calipso', 'bootstrap-markdown', 'component/backbone-forms-editor-markdown', 
         'component/backgrid-edit-in-modal-button-cell'],
		function( Calipso,  Markdown, BackboneFormMarkdown, EditInModalCell) {
			//Backbone.Form.editors.Markdown
			var TextModel = ResourceModel.extend({
				modelKey: "text",
			},
			// static members
			{
				parent: ResourceModel
			});

			/**
			 * Get the model class URL fragment corresponding this class
			 * @returns the URL path fragment as a string
			 */
			TextModel.prototype.getPathFragment = function() {
				return "texts";
			}
			TextModel.prototype.getFormSchemas = function() {
				// superclass schema
				var superSchema = TextModel.parent.getFormSchemas();
				// own schema
				var schema = {
					"sourceContentType" : {
						"search": { type: 'Select', options: [ 'text/plain', 'text/x-markdown', 'text/html' ] },
						"default": { 
							type: 'Select', 
							options: [ 'text/plain', 'text/x-markdown', 'text/html' ], 	
							validators : [ 'required' ]
						}
					},
					"source" : {
						"search": 'Text',
						"default": {
							type: BackboneFormMarkdown,
							validators : [ 'required' ],
							editorAttrs: { "data-provide": 'markdown' }
						},
					}
					
				};
				// return merged schemas
				return $.extend({}, superSchema, schema);
			}
			
			TextModel.prototype.getGridSchema = function() {
				var superSchema = ResourceModel.prototype.getGridSchema();
				var localSchema = [/*{
					name : "name",
					label : "name",
					cell : "string"
				}*/];
				// return merged schemas
				return $.extend({}, superSchema, localSchema);
			}
			return TextModel;
		});

File: calipso-hub-core/src/main/java/gr/abiss/calipso/model/cms/Text.java
/**
 * Copyright (c) 2007 - 2013 www.Abiss.gr
 *
 * This file is part of Calipso, a software platform by www.Abiss.gr.
 *
 * Calipso is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Calipso is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with Calipso. If not, see http://www.gnu.org/licenses/agpl.html
 */
package gr.abiss.calipso.model.cms;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

/**
 */
@Entity
@Table(name = "content_text")
public class Text extends Resource {
	
	public static final String MIME_MARKDOWN = "text/x-markdown";

	/**
	 * The MIME type of the page source. May be used to dynamically generate response content. 
	 */
	@Column(name = "source_content_type", nullable = false)
	private String sourceContentType;
	
	/**
	 * The raw text content of the resource 
	 */
	@Column(name = "source", nullable = false)
	private String source;

	public String getSourceContentType() {
		return sourceContentType;
	}

	public void setSourceContentType(String sourceContentType) {
		this.sourceContentType = sourceContentType;
	}

	public String getSource() {
		return source;
	}

	public void setSource(String source) {
		this.source = source;
	}

	
}